### PR TITLE
Non-human mobs can now benefit from held id cards and economy.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Moved held items checks of `get_idcard()` et similars to `/mob/living`. Removed said getters from `/mob` since camera and dead mobs don't need them. Removed checks that still needlessly restricted these getters to humans, and added several `isliving` ones now that the getters are no longer defined on `/mob`.

## Why It's Good For The Game
There are many features that are currently restricted to human mobs ~not because monkeys/critters are neither natural, legal or juridical people, but~ only cause the code was defined on that level.

## Changelog
:cl:
tweak: Non-human mobs now also benefit from held id cards and some economy features (dexterity checks still apply).
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -531,3 +531,32 @@
...
+	if(!length(held_items)) //Early return for mobs without hands.
+		return
+	//Check hands
+	var/obj/item/held_item = get_active_held_item()
+	if(held_item) //Check active hand
+		. = held_item.GetID()
+	if(!.) //If there is no id, check the other hand
+		held_item = get_inactive_held_item()
+		if(held_item)
+			. = held_item.GetID()
```
Comment (Rohesie) [inline code/modules/mob/mob_helpers.dm:546]: Getters are pretty bad and we shouldn't expand on them.
The proper way would be to retrieve the active hand held item and have the information on whether that is currently acting as an ID through a variable in it, which is modified accordingly as the events change.
That would be a bigger refactor, though, and not necessary here. Just keep it in mind for future changes.

That said, could you put these getters in the `/mob/living` level? We don't need non-living mobs to check for ID, do we?

**Comment by RaveRadbury** [comment]
I can't wait to watch monkeys buy packs of smokes and smoke them.

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/mob/mob_helpers.dm]: Done.

**Comment by Rohesie** [comment]
Tested and working?

**Comment by ATH1909** [comment]
this looks like it'll be really helpful if I decide to give service borgs the ability ID cards so that they can collect tips and spend cash at vending machines later.

**Comment by Ghommie** [comment]
Tested and working. Monkey still need dexterity (jungle fever) to use most machines.

**Comment by spookydonut** [comment]
@Ghommie rebase/merge commit for working CI

**Pull request merged**